Upstream-Status: Inapproriate

Remove flag for DSE MemorySSA as it is a feature available in the
newest clang llvm revisions. Not all the yocto releases use the newest
one. Otherwise, clang will complain about unknown command line argument.
This should be revisited when our supported yocto branches have newer
clang version.

Signed-off-by: Maksim Sisov <msisov@igalia.com>
---
diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index 85fbbbe614bf..fba9fd10426e 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -489,25 +489,6 @@ config("compiler") {
       "-mllvm",
       "-instcombine-lower-dbg-declare=0",
     ]
-
-    # Pinned clang has enable-dse-memoryssa enabled by default but it's broken,
-    # so we have to force it off.
-    # Trunk clang has it disabled by default but it might work, so we force it
-    # on so the ToT bots can check if it works now.
-    if (!llvm_force_head_revision) {
-      cflags += [
-        # TODO(https://crbug.com/1127713): Investigate, remove.
-        "-mllvm",
-        "-enable-dse-memoryssa=false",
-      ]
-    }
-    if (llvm_force_head_revision) {
-      cflags += [
-        # TODO(https://crbug.com/1127713): Investigate, remove.
-        "-mllvm",
-        "-enable-dse-memoryssa=true",
-      ]
-    }
   }
-- 
2.25.1
